chain/events/filter/index.go

```diff
@@ -557,18 +658,26 @@ func (ei *EventIndex) prefillFilter(ctx context.Context, f *eventFilter, exclude
 		}
 	}
 
-	if excludeReverted {
-		clauses = append(clauses, "event.reverted=?")
-		values = append(values, false)
+	// Resolve emitters from filter addresses at the latest TipSet.
+	// TODO investigate if it is safe to use the current tipset; what about finality?
```

Comment:
Can you elaborate on the concern here? I would imagine that for prefill that you just want a best guess, and current head is that best guess and should represent the correct state as far as we know it up until this point. Is there even a reasonable alternative to doing this?

Reply:
OK, so the alternative you're suggesting would be to rewind to finality and resolve on that?

I think we might be getting into the realm of guessing user-intention here unfortunately and we don't have enough signal for that—we'd probably need a "resolve tipset" to get it exactly the way they intended. I think as long as we're doing it consistently against the latest tipset then we should be pretty likely to get what the user wants. Although my experience doesn't give me any useful hypotheticals to test that theory against! If it ever comes up as a complaint then maybe we consider the options then?